src/main/kotlin/org/jetbrains/tinygoplugin/services/ErrorRegexUtil.kt

```diff
@@ -0,0 +1,71 @@
+package org.jetbrains.tinygoplugin.services
+
+import com.goide.sdk.GoSdkService
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import org.jetbrains.tinygoplugin.TinyGoBundle
+
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_UNKNOWN = "tinygoSDK.incompatibleGoVersion.unknownVersion"
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_TITLE = "tinygoSDK.incompatibleGoVersion.title"
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_RANGE_SPECIFIED_MESSAGE =
+    "tinygoSDK.incompatibleGoVersion.rangeSpecified.message"
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_RANGE_UNSPECIFIED_MESSAGE =
+    "tinygoSDK.incompatibleGoVersion.rangeUnspecified.message"
+
+fun generateMessageIfVersionErrorFound(project: Project, processOutput: String): String? {
+    val incompatibleGoVersionsMatch = extractIncompatibleVersionsError(processOutput)
+    val couldNotReadGoVersionMatch = extractCouldNotReadGoVersionError(processOutput)
```

Comment:
The result of this method is never used if the call above returns not `null`. It's not a big deal here as this method is only called on a non-zero exit code. But in general, one should try to avoid unnecessary executions.